### DISCUSSION TITLE
FieldState.invalid and fieldState.error are inconsistent

### DISCUSSION THREAD

**Discussion by SebastianStehle** [react-hook-form/react-hook-form#9539] [Q&A]
Hi,

I have a simple error component:

```
export const Error = ({ name }: { name: string }) => {
   const { fieldState, formState } = useController({ name });

  return (
    <FormControlError error={fieldState.error?.message} submitCount={formState.submitCount} touched={fieldState.isTouched} />
  );
};
```

The logic is that the error is shown if there is an error message and either submitCount is greater than zero or touched is set to true.

But before I press submit, `fieldState.invalid == true` and `fieldState.error == undefined`. It is very weird.

**Comment by SebastianStehle** [comment]
Sorry, all my fault...

**Reply by bluebill1049** [reply]
curious to know what was the issue?
